### PR TITLE
Feature/profile dropdown

### PR DESCRIPTION
**Pull request by Dhruvi-Makwana**

### CODE REVIEW

File: Chatify/chat/models.py [hunk 1]
```diff
@@ -6,3 +6,4 @@
 class User(AbstractUser):
     profile_photo = models.ImageField(upload_to="profile_photo/")
     mobile_number = PhoneNumberField(unique=True)
+    online = models.BooleanField(default=False)
```
Comment (yatishTrootech) [inline Chatify/chat/models.py]: add 1 line only

File: Chatify/chat/static/js/chat.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            if (currentUser) {
+            currentUser.messages.sender.push({
+                  user: currentUser.name,
+                  profile: currentUser.profile,
+                  message: message,
+            });
+          }
+        }
+
+          $scope.status = 'online';
```
Comment (yatishTrootech) [inline Chatify/chat/static/js/chat.js]: All should be on top $scope.status = 'online';

File: Chatify/chat/static/js/chat.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+          $scope.setStatus = function(status) {
+          console.log(status)
```
Comment (yatishTrootech) [inline Chatify/chat/static/js/chat.js]: proper indentation

File: Chatify/chat/static/js/chat.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+          $scope.status = status;
+          }
```
Comment (yatishTrootech) [inline Chatify/chat/static/js/chat.js]: remove console

File: Chatify/chat/static/js/register.js [hunk 5]
```diff
@@ -1,9 +1,10 @@
-$(document).on("submit", "#register_form", function(event) {
-        console.log("hii")
+
+ $(document).on("submit", "#register_form", function(event) {
```
Comment (yatishTrootech) [inline Chatify/chat/static/js/register.js]: wrap up in a function don't register events

File: Chatify/chat/templates/chat/base.html [hunk 6]
```diff
@@ -3,30 +3,25 @@
...
-        <script src="{% static 'js/register.js' %}"></script>
-        <script src="{% static 'js/bootstrap.bundle.min.js' %}"></script>
-        <script src="{% static 'js/login.js' %}"></script>
+        <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.4/jquery.min.js"></script>
+        <script type="text/javascript"
+        src="{% static 'js/jquery.mCustomScrollbar.min.js' %}"></script>
+        <script src="{% static 'js/chat.js' %}"></script>
+        <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/js/bootstrap.bundle.min.js"></script>
+
+        <link rel="stylesheet" href="{% static 'css/register.css' %}">
```
Comment (yatishTrootech) [inline Chatify/chat/templates/chat/base.html]: <link rel="stylesheet" href="{% static 'css/register.css' %}"> should be in register html page

File: Chatify/chat/templates/chat/register.html [hunk 7]
```diff
@@ -1,39 +1,42 @@
...
-    <div class="field">
-      <input type="text" name="mobile_number" required>
-      <label>Mobile</label>
-    </div>
-    <div class="field">
-      <input type="file" class="pt-2" name="profile_photo" required>
-    </div>
+             {% block content %}
+ <script src="{% static 'js/register.js' %}"></script>
+ <script src="{% static 'js/ajax_request.js' %}"></script>
```
Comment (yatishTrootech) [inline Chatify/chat/templates/chat/register.html]: <script src="{% static 'js/ajax_request.js' %}"></script> should be in base.js

**Review by yatishTrootech — CHANGES_REQUESTED** [review]

File: Chatify/chat/api.py [hunk 8]
```diff
@@ -28,3 +28,13 @@ def get(self, request):
             return Response(
                 {"errors": str(e.detail)}, status=status.HTTP_400_BAD_REQUEST
             )
+
+
+class UserOnlineAPIView(APIView):
+    def put(self, request, *args, **kwargs):
```
Comment (yatishTT) [inline Chatify/chat/api.py:34]: def post

File: Chatify/chat/api.py [hunk 9]
```diff
@@ -28,3 +28,13 @@ def get(self, request):
...
+        user = request.user.id
+        if request.data.get("online") == "true":
```
Comment (yatishTT) [inline Chatify/chat/api.py]: user.save(online=True if request.data.get("online") else False)

File: Chatify/chat/api.py [hunk 10]
```diff
@@ -28,3 +28,13 @@ def get(self, request):
...
+            user.save(online=True)
+        else:
+            user.save(online=False)
+        return Response(status=status.HTTP_400_BAD_REQUEST)
```
Comment (yatishTT) [inline Chatify/chat/api.py]: why 400 on success ?

File: Chatify/chat/models.py [hunk 11]
```diff
@@ -6,3 +6,4 @@
 class User(AbstractUser):
     profile_photo = models.ImageField(upload_to="profile_photo/")
     mobile_number = PhoneNumberField(unique=True)
+    online = models.BooleanField(default=False, blank=True, null=True)
```
Comment (yatishTT) [inline Chatify/chat/models.py]: boolean flag starts from is > is_online

File: Chatify/chat/serializers.py [hunk 12]
```diff
@@ -18,6 +18,7 @@ class Meta:
             "mobile_number",
             "password",
             "confirm_password",
+            "online",
```
Comment (yatishTT) [inline Chatify/chat/serializers.py:21]: is_online

File: Chatify/chat/static/css/chat.css [hunk 13]
```diff
@@ -270,3 +270,15 @@ body,
         {
           background: #00000052;
         }
+        .profile-dropdown
+        {
+        background-color: rgba(0, 0, 0, 0.4) !important;
+        border:none;
+        }
```
Comment (yatishTT) [inline Chatify/chat/static/css/chat.css:277]: wrong indentation

File: Chatify/chat/templates/chat/chat.html [hunk 14]
```diff
@@ -2,10 +2,23 @@
...
+
 <script src="{% static 'js/chat.js' %}"></script>
 
 
     <div class="container-fluid h-100" ng-app="ChatApp" ng-controller="chatCtrl">
+            <!-- porofile dropdown -->
+           <div class="dropdown d-flex flex-row-reverse mt-3">
+                <button type="button" class="btn btn-primary dropdown-toggle profile-dropdown" data-bs-toggle="dropdown">
+                    <img src="http://127.0.0.1:8000/static/images/profile.png" height="20px" width="20px" class="rounded-circle m-2 pe-2">
+                        <span>Dhruvi</span>
```
Comment (yatishTT) [inline Chatify/chat/templates/chat/chat.html:15]: why name and DP is static ?

**Review by yatishTT — CHANGES_REQUESTED** [review]

**Pull request merged**